src/ServiceControl.Monitoring/Metrics/Raw/LongValueOccurenceSerializerDefinition.cs

```diff
@@ -24,33 +24,52 @@ public void Serialize(object message, Stream stream)
 
         public object[] Deserialize(Stream stream, IList<Type> messageTypes = null)
         {
-            var message = new LongValueOccurrences();
+            var reader = new BinaryReader(stream);
 
-            ToLongValueOccurrences(message, stream);
+            var version = reader.ReadInt64();
 
-            return new object[]
+            if (version == 1)
             {
-                message
-            };
-        }
+                var baseTicks = reader.ReadInt64();
+                var count = reader.ReadInt32();
 
-        static void ToLongValueOccurrences(LongValueOccurrences message, Stream stream)
-        {
-            var reader = new BinaryReader(stream);
+                if (count == 0)
+                {
+                    // no records, return empty
+                    return new object[0];
+                }
 
-            message.Version = reader.ReadInt64();
-            message.BaseTicks = reader.ReadInt64();
+                var messages = new List<object>(1); // usual case
+                var message = LeaseMessage(messages);
 
-            var count = reader.ReadInt32();
+                for (var i = 0; i < count; i++)
+                {
+                    var ticks = reader.ReadInt32();
+                    var value = reader.ReadInt64();
+                    var date = baseTicks + ticks;
 
-            message.Ticks = new int[count];
-            message.Values = new long[count];
+                    if (message.TryRecord(date, value) == false)
+                    {
+                        message = LeaseMessage(messages);
```

Comment:
@mikeminutillo I agree it is a bit unfortunate that there is no symmetry but I'm afraid there is not much we can do here since the serializer doesn't allow to float things from the pipeline into the serializer. In order to have symmetry, you'd need a behavior which gets the lease and then passes that into the serializer and then releases the lease after the pipeline returned. I see no way without ambient context to achieve it. So instead of introducing an ambient context I would stay with the current approach.